### DISCUSSION TITLE
Is it possible to update HTTPPassword using gok update?

### DISCUSSION THREAD

**Discussion by nhanb** [gokrazy/gokrazy#200] [Q&A] [answered]
There's a chicken-and-egg problem: if I update the `Update -> HTTPPassword` field in config.json and run `gok update`, it fails to authenticate in the first place: `unexpected HTTP status code: got 401, want 200 (body "invalid username/password\n")`, which, admittedly, makes sense.

So, is my only option to plug the gokrazy drive back into my development laptop and run `gok overwrite --full /dev/sdX` instead?

I'm sorry if this is obvious; I'm still trying to figure out the gokrazy workflow.

**Comment by stapelberg** [comment, marked as answer]
> So, is my only option to plug the gokrazy drive back into my development laptop and run `gok overwrite --full /dev/sdX` instead?

That is certainly the easiest way.

Another possibility is to specify the old update flag in your config.json:

```json
    "InternalCompatibilityFlags": {
        "Update": "http://gokrazy:oldpassword@hostname"
    }
```

Maybe we should consider adding a separate flag to [the Update struct](https://gokrazy.org/userguide/instance-config/#update)

**Reply by christian-heusel** [reply]
> Maybe we should consider adding a separate flag to [the Update struct](https://gokrazy.org/userguide/instance-config/#update)

Wouldnt it make more sense to have this as a parameter (something like `--new-password="..."`) that then overwrites the value in `config.json` on successfull update?

**Reply by stapelberg** [reply]
> Wouldnt it make more sense to have this as a parameter (something like `--new-password="..."`) that then overwrites the value in `config.json` on successfull update?

We generally refrain from updating config.json where possible, but yes, it could also be a flag (maybe --current-password).

**Reply by nhanb** [reply]
Thanks @stapelberg! The `InternalCompatibilityFlags` trick worked for me.

> Maybe we should consider adding a separate flag to [the Update struct](https://gokrazy.org/userguide/instance-config/#update)

You mean something like `Update.OldHTTPPassword`? I'm for that as well.

Regardless of the solution, I'd always prefer `gok update` over `gok overwrite` simply because the latter forces me to rip the storage drive out of my appliance, plug it into my laptop, then put it back in the appliance. It's not a big deal for an mSD card inside a Raspberry Pi, but it's a pain when dealing with, say, a SATA drive inside a mini PC. I find it very nice to be able to close up the appliance PC and continue developing/deploying over LAN.
